benchmark/fs/readfile.js

```diff
@@ -23,28 +24,34 @@ function main(conf) {
   data = null;
 
   var reads = 0;
-  var bench_ended = false;
+  var benchEnded = false;
   bench.start();
   setTimeout(function() {
-    bench_ended = true;
+    benchEnded = true;
+    console.log(`bench ended: reads ${reads}`);
     bench.end(reads);
     try { fs.unlinkSync(filename); } catch (e) {}
-    process.exit(0);
   }, +conf.dur * 1000);
 
   function read() {
     fs.readFile(filename, afterRead);
   }
 
   function afterRead(er, data) {
-    if (er)
+    if (er) {
+      if (er.code === 'ENOENT') {
+        // Race with timer.
```

Comment:
can you please expand this comment a bit more, why is it racing?

Reply:
See new explanatory comment in f4dc004.

In this benchmark, `main()` terminates execution after 5 seconds by setting `benchEnded` and unlinking the test file. This is a file system race with the `read` jobs, which may see ENOENT.

This benchmark was flaky before this PR -- it would occasionally error out because there was no test for `benchEnded`. I added it in the original `readfile.js` as well as `readfile-partitioned.js`.